core/src/scene/filters.h

```diff
@@ -57,19 +62,29 @@ struct Filter {
     Filter() : data(none_type{}) {}
     Filter(Data _data) : data(std::move(_data)) {}
 
+    static std::vector<Filter> sort(const std::vector<Filter>& filters);
+    void print(int _indent = 0) const;
+    int matchCost() const;
+    const std::string& key() const;
+    const std::vector<Filter>& operands() const;
+
     // Create an 'any', 'all', or 'none' filter
     inline static Filter MatchAny(const std::vector<Filter>& filters) {
-        return { OperatorAny{ filters }};
+        return { OperatorAny{ sort(filters) }};
     }
     inline static Filter MatchAll(const std::vector<Filter>& filters) {
-        return { OperatorAll{ filters }};
+        return { OperatorAll{ sort(filters) }};
     }
     inline static Filter MatchNone(const std::vector<Filter>& filters) {
-        return { OperatorNone{ filters }};
+        return { OperatorNone{ sort(filters) }};
     }
     // Create an 'equality' filter
     inline static Filter MatchEquality(const std::string& k, const std::vector<Value>& vals) {
+        // if (vals.size() == 1) {
+        //     return { EqualityOne{ k, vals[0], globalType(k) }};
+        // } else {
```

Comment:
Couldn't see a difference. I guess it wouldn't  hurt to add it back though - just was one case more to handle during the cost evaluation..

I'll go over it for cleanups tomorrow
 over it for cleanups tomorrow


I'll go over it for cleanups tomorrow
, just was one more case to handle during 

I'll go over it for cleanups tomorrow